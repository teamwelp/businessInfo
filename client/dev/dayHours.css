.row {
  display: flex;
}
.day {
  width: 50px;
  font-weight: bold;
}
.dayHoursBox {
  display: flex;
  align-items: flex-end;
}
.isOpen {
  color: red;
  font-weight: bold;
  font-size: 13px;
  margin-left: 8px;
}