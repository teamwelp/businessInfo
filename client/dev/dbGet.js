import axios from 'axios';

export default function () {
  const url = window.location.href.slice(window.location.href.search('biz') + 4).replace('/', '');
  axios.get(`http://127.0.0.1:9001/id/${url}/`)
    .then((response) => {
      this.setState({ data: response.data[0] });
    })
    .catch((error) => {
      console.log(error);
    });
};
