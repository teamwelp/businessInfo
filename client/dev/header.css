.header {
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  color: #333;
  background-color: #f5f5f5;
  width: 520px;
  height: 80px;
  margin-left: 150px;
}
.name {
  float: left;
  margin: 10px 10px 0px 10px;
  font-size: 30px;
  font-weight: bold;
}
.claimdiv {
  float: left;
  margin: 24px 0px 0px 0px;
  font-size: 15px;
}
.claimicon {
  float: left;
  position: relative;
  bottom: 3px;
  color: #0073bb;
  margin-right: 5px;
}
.unclaimed {
  float: left;
  color: #0073bb;
  text-decoration: none;
}
.claimed {
  float: left;
  color: #333;
}
.bottomHeader {
  float: left;
  margin: 10px;
  font-size: 15px;
  width: 400px;
}
.bullet {
  display: inline;
  color: #666;
  margin-left: 10px;
  margin-right: 10px;
}
.metatags {
  display: inline;
  font-weight: bold;
}
.contact {
  margin-left: 150px;
  width: 300px;
  border: 1px solid #999;
  background-color: #ffffe6;
  padding: 4px;
}
.grayicon {
  float: left;
  clear: left;
  color: #777;
  margin-right: 8px;
  position: relative;
  top: 2px;
}
.addressPhone {
  float: left;
  margin: 4px 0 10px 0;
  line-height: 1.25;
}
.contactLink {
  float: left;
  color: #0073bb;
  text-decoration: none;
  margin: 4px 0 10px 0;
  line-height: 1.25;
}
.clearBoth {
  clear: both;
}