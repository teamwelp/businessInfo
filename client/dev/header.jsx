import FaQuestion from 'react-icons/lib/fa/question-circle';
import FaCheck from 'react-icons/lib/fa/check-circle';
import FaLocation from 'react-icons/lib/fa/map-marker';
import FaLink from 'react-icons/lib/fa/external-link';
import FaPhone from 'react-icons/lib/fa/phone';
import FaCellPhone from 'react-icons/lib/fa/mobile';
import React from 'react';
import styles from './header.css';

const calcClaimed = (props) => {
  let claimedContent = 'Unclaimed';
  let claimIcon = <FaQuestion />;
  let claimStyle = styles.unclaimed;
  if (props.data.claimedByOwner) {
    claimedContent = 'Claimed';
    claimIcon = <FaCheck />;
    claimStyle = styles.claimed;
  }
  return {
    claimedContent,
    claimIcon,
    claimStyle,
  };
};
const calcPrice = (props) => {
  let priceScale = '';
  for (let i = 0; i < props.data.priceRangeScale; i += 1) {
    priceScale = priceScale.concat('$');
  }
  return priceScale;
};

export default function (props) {
  return (
    <div>
      <div className={styles.header}>
        <div className={styles.name}>{props.data.name}</div>
        <div className={styles.claimdiv}>
          <div className={styles.claimicon}>{calcClaimed(props).claimIcon}</div>
          <div className={calcClaimed(props).claimStyle}>{calcClaimed(props).claimedContent}</div>
        </div>
        <div className={styles.bottomHeader}>
          {calcPrice(props)}
          <div className={styles.bullet}>&bull;</div>
          <div className={styles.metatags}>{props.data.metatags.join(', ')}</div>
        </div>
      </div>
      <div className={styles.contact}>
        <div className={styles.address}>
          <div className={styles.grayicon}><FaLocation /></div>
          <div className={styles.addressPhone}>
            {props.data.addressNumber} {props.data.addressStreet} 
            <br />
            {props.data.addressCity}, {props.data.addressState} {props.data.addressZip}
          </div>
        </div>
        <div>
          <div className={styles.grayicon}><FaPhone /></div>
          <div className={styles.addressPhone}>
            ({props.data.phoneAreaCode}) {props.data.phoneOfficeCode}-{props.data.phoneLineCode}
          </div>
        </div>
        <div>
          <div className={styles.grayicon}><FaLink /></div>
          <a className={styles.contactLink} href={props.data.businessLink}>
            {props.data.businessLink}
          </a>
        </div>
        <div>
          <div className={styles.grayicon}><FaCellPhone /></div>
          <a className={styles.contactLink} href="#">Send to your Phone</a>
        </div>
        <div className={styles.clearBoth} />
      </div>
    </div>
  );
}
