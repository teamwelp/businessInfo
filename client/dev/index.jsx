import React from 'react';
import ReactDOM from 'react-dom';
import axios from 'axios';
import styles from './index.css';
import Header from './header';
import Sidebox from './sidebox';
import SideHours from './sideHours';

const starterData = { '_id': '5a95c36256028a0cd03c9307', 'carParking': ['Parking Lot', 'Garage, Validated'], 'metatags': ['Soul Food', 'Bars', 'Dinner', 'Brunch'], 'acceptsCreditCards': false, 'bikeParking': true, 'byApptOnly': true, 'claimedByOwner': true, 'goodForKids': true, 'isYelpAdvertiser': true, 'id': 200, 'phoneAreaCode': 415, 'addressNumber': 301, 'healthInspection': 88, 'phoneLineCode': 7940, 'phoneOfficeCode': 421, 'priceRangeLow': 29, 'priceRangeRange': 9, 'priceRangeScale': 1, 'hours': { 'Mon': { 'open': 9, 'close': 18, '_id': '5a95c36256028a0cd03c92ff' }, 'Tue': { 'open': 9, 'close': 18, '_id': '5a95c36256028a0cd03c9300' }, 'Wed': { 'open': 9, 'close': 18, '_id': '5a95c36256028a0cd03c9301' }, 'Thu': { 'open': 9, 'close': 18, '_id': '5a95c36256028a0cd03c9302' }, 'Fri': { 'open': 9, 'close': 18, '_id': '5a95c36256028a0cd03c9303' }, 'Sat': { 'open': 10, 'close': 21, '_id': '5a95c36256028a0cd03c9304' }, 'Sun': { 'open': 10, 'close': 21, '_id': '5a95c36256028a0cd03c9305' }, '_id': '5a95c36256028a0cd03c9306' }, 'businessLink': 'http://olegsburgerpalace.com', 'addressCity': 'San Francisco', 'addressState': 'CA', 'addressStreet': 'Mission Street', 'addressZip': '94103', 'longDescription': 'Nostrud do et commodo adipisicing sunt cupidatat voluptate duis proident est laborum. Culpa reprehenderit incididunt proident magna Lorem adipisicing deserunt quis sint eiusmod tempor esse. Culpa culpa amet exercitation proident velit in culpa eu commodo culpa. Aliquip cillum veniam consectetur laboris quis proident ullamco proident ullamco anim sint labore non elit. Minim sunt cupidatat tempor exercitation consectetur. Ipsum minim velit ex tempor enim elit ad. Duis dolore mollit commodo ea Lorem reprehenderit sint ex eu cillum. Enim voluptate dolore in esse occaecat pariatur anim reprehenderit commodo est pariatur dolor pariatur aliquip. Nulla et amet sunt ex consequat exercitation eiusmod amet non. Quis Lorem non anim deserunt.', 'name': 'Oleg\'s Burger Palace', '__v': 0 };

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: starterData,
    };
  }
  componentWillMount() {
    let url = document.location.href.split('/');
    url = url[url.length - 2];
    axios.get(`http://127.0.0.1:3000/id/${url}/`)
      .then((response) => {
        this.setState({ data: response.data[0] });
      })
      .catch((error) => {
        console.log(error);
      });
  }
  render() {
    return (
      <div>
        <Header data={this.state.data} />
        <div className={styles.sidebar}>
          <Sidebox data={this.state.data} />
          <SideHours data={this.state.data} />
        </div>
      </div>
    );
  }
}

ReactDOM.render(<App />, document.getElementById('app'));
