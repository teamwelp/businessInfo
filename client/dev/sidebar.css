.sidebar {
  float: right;
  width: 300px;
  margin-right: 150px;
}