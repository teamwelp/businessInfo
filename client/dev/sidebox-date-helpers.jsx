import React from 'react';

const getTodaysHours = (hours) => {
  const daysOfWeek = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
  const today = new Date(Date.now());
  const day = daysOfWeek[today.getDay()];
  if (hours[day].open) return hours[day];
  return false;
};

const getTodaysHoursContent = (todaysHours) => {
  if (!todaysHours) return false;
  let { open, close } = todaysHours;
  let openSuffix = 'am';
  let closeSuffix = 'am';
  if (open > 12) {
    open -= 12;
    openSuffix = 'pm';
  }
  if (close > 12) {
    close -= 12;
    closeSuffix = 'pm';
  }
  return `${open}:00 ${openSuffix} - ${close}:00 ${closeSuffix}`;
};

const RenderHoursToday = (props) => {
  if (props.todaysHours) {
    const hoursTodayContent = getTodaysHoursContent(props.todaysHours);
    return <span>Today <b>{hoursTodayContent}</b></span>;
  }
  return <span>Closed today</span>;
};

const isOpen = (todaysHours) => {
  if (!todaysHours) return false;
  const today = new Date(Date.now());
  const open = new Date(today.getFullYear(), today.getMonth(), today.getDate(), todaysHours.open);
  const close = new Date(today.getFullYear(), today.getMonth(), today.getDate(), todaysHours.close);
  if (today.valueOf() >= open.valueOf() && today.valueOf() <= close.valueOf()) {
    return true;
  }
  return false;
};

export { getTodaysHours, getTodaysHoursContent, RenderHoursToday, isOpen };
