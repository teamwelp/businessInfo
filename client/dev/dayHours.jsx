import React from 'react';
import styles from './dayHours.css';
import { getTodaysHoursContent, isOpen } from './sidebox-date-helpers';

const getHours = (props) => {
  if (props.day === undefined) return 'Closed';
  return getTodaysHoursContent(props.day);
};

const getIsOpen = (props) => {
  const getIsOpenToday = () => {
    let todayInteger = new Date(Date.now()).getDay();
    if (todayInteger === 0) {
      todayInteger = 6;
    } else {
      todayInteger -= 1;
    }
    const sortedDays = ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'];
    if (todayInteger === sortedDays.indexOf(props.day.name)) return true;
    return false;
  };
  if (isOpen(props.day) && getIsOpenToday()) return 'Open now';
  return '';
};

export default (props) => {
  return (
    <div className={styles.row}>
      <div className={styles.day}>{props.day.name}</div>
      <div className={styles.dayHoursBox}>
        <div className={styles.dayHours}>{getHours(props)}</div>
        <div className={styles.isOpen}>{getIsOpen(props)}</div>
      </div>
    </div>
  );
};
