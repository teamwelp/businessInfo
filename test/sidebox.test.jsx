import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import puppeteer from 'puppeteer';
import renderer from 'react-test-renderer';
import Sidebox from '../client/dev/sidebox';
import starterData from './starterdata';
import { getTodaysHours, getTodaysHoursContent, RenderHoursToday } from '../client/dev/sidebox-date-helpers';

const { shallow } = Enzyme;
Enzyme.configure({ adapter: new Adapter() });
Date.now = jest.fn(() => new Date('June 4, 2018').valueOf());

describe('Sidebox Helpers Unit Test', () => {
  test('return correct hours for getTodaysHours on Monday', () => {
    expect(getTodaysHours(starterData.hours).open).toBe(9);
    expect(getTodaysHours(starterData.hours).close).toBe(18);
  });
  test('return correct formatted string on Monday', () => {
    const todaysHours = getTodaysHours(starterData.hours);
    expect(getTodaysHoursContent(todaysHours)).toBe('9:00 am - 6:00 pm');
  });
  test('render correct hours on Monday test data', () => {
    const todaysHours = getTodaysHours(starterData.hours);
    const wrapper = shallow(<RenderHoursToday todaysHours={todaysHours} />);
    expect(wrapper.find('span').html()).toBe('<span>Today <em>9:00 am - 6:00 pm</em></span>');
  });
  test('if closed today, renders correct message', () => {
    const newStart = Object.assign(starterData);
    delete newStart.hours.Mon.open;
    const todaysHours = getTodaysHours(newStart.hours);
    const wrapper = shallow(<RenderHoursToday todaysHours={todaysHours} />);
    expect(wrapper.find('span').html()).toBe('<span>Closed today</span>');
  });
});
