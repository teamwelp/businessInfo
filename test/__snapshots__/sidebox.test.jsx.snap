// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Component snapshot test 1`] = `
<div
  className="sidebox"
>
  <div
    className="categoryBox"
  >
    <svg
      className="icon clockOpen"
      fill="currentColor"
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "color": undefined,
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 40 40"
      width="1em"
    >
      <g>
        <path
          d="m23 12.1v10q0 0.4-0.2 0.6t-0.5 0.2h-7.2q-0.3 0-0.5-0.2t-0.2-0.6v-1.4q0-0.3 0.2-0.5t0.5-0.2h5v-7.9q0-0.3 0.2-0.5t0.6-0.2h1.4q0.3 0 0.5 0.2t0.2 0.5z m9.3 7.9q0-3.3-1.6-6.1t-4.5-4.4-6.1-1.6-6.1 1.6-4.4 4.4-1.6 6.1 1.6 6.1 4.4 4.4 6.1 1.6 6.1-1.6 4.5-4.4 1.6-6.1z m5 0q0 4.7-2.3 8.6t-6.3 6.2-8.6 2.3-8.6-2.3-6.2-6.2-2.3-8.6 2.3-8.6 6.2-6.2 8.6-2.3 8.6 2.3 6.3 6.2 2.3 8.6z"
        />
      </g>
    </svg>
    <div
      className="contentBox column"
    >
      <span
        className="hoursToday"
      >
        <span>
          Today 
          <b>
            9:00 am - 6:00 pm
          </b>
        </span>
      </span>
      <span
        className="openNow"
      >
        <b>
          Open now
        </b>
      </span>
    </div>
  </div>
  <span
    className="hl"
  />
  <div
    className="categoryBox"
  >
    <svg
      className="icon"
      fill="currentColor"
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "color": undefined,
          "verticalAlign": "middle",
        }
      }
      viewBox="0 0 40 40"
      width="1em"
    >
      <g>
        <path
          d="m24.8 19.2l-2.5 2.4 11.5 11.5-2.3 2.4-11.5-11.5-11.5 11.5-2.3-2.4 16.2-16.2q-0.9-1.9-0.3-4.4t2.7-4.5q2.4-2.4 5.4-2.7t4.7 1.4 1.5 4.8-2.8 5.5q-2 1.9-4.5 2.5t-4.3-0.3z m-11.3 3.1l-7-7.1q-2-1.9-2-4.7t2-4.6l11.7 11.6z"
        />
      </g>
    </svg>
    <div
      className="contentBox"
    >
      <a
        className="boldLink"
        href="https://www.yelp.com/menu/gary-danko-san-francisco"
      >
        Full menu
      </a>
    </div>
  </div>
  <span
    className="hl"
  />
  <div
    className="categoryBox"
  >
    <div
      className="icon"
    >
      <div
        className="priceIcon"
      >
        <div
          className="priceIconRed"
        >
          $
        </div>
        <div
          className="priceIconGray"
        >
          $$$
        </div>
      </div>
    </div>
    <span
      className="contentBox"
    >
      Price Range  
      <b>
        $29-38
      </b>
    </span>
  </div>
  <span
    className="hl"
  />
  <div
    className="categoryBox"
  >
    <div
      className="icon flexCenter"
    >
      <div
        className="healthIcon"
      >
        88
      </div>
    </div>
    <div
      className="contentBox"
    >
      <a
        className="boldLink"
        href="https://www.yelp.com/inspections/gary-danko-san-francisco"
      >
        Health Inspection
      </a>
      <span
        className="healthtext"
      >
         
        88
         out of 100
      </span>
    </div>
  </div>
</div>
`;
